src/main/java/net/ccbluex/liquidbounce/injection/mixins/minecraft/gui/MixinPlayerListHud.java

```diff
@@ -18,17 +18,180 @@
+    }
+
+    @ModifyExpressionValue(method = "render", at = @At(
+            value = "FIELD",
+            target = "Lnet/minecraft/client/gui/hud/PlayerListHud$ScoreDisplayEntry;name:Lnet/minecraft/text/Text;"
+    ))
+    private Text hookVisibilityName(Text original, @Local(ordinal = 0) PlayerListEntry entry) {
+        if (!ModuleBetterTab.INSTANCE.getRunning()) {
+            return original;
+        }
+
+        return ModuleBetterTab.Visibility.INSTANCE.getNameOnly() ?
+                Text.of(entry.getProfile().getName()) : original;
+
+    }
+
+    @ModifyExpressionValue(method = "render", at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/client/gui/hud/PlayerListHud;getPlayerName(Lnet/minecraft/client/network/PlayerListEntry;)Lnet/minecraft/text/Text;"
+    ))
+    private Text hookWidthVisibilityName(Text original, @Local(ordinal = 0) PlayerListEntry entry) {
+        if (!ModuleBetterTab.INSTANCE.getRunning()) {
+            return original;
+        }
+
+        return ModuleBetterTab.Visibility.INSTANCE.getNameOnly() ?
+                Text.of(entry.getProfile().getName()) : original;
+    }
+
+    @Inject(method = "render", at = @At(value = "INVOKE", target = "Ljava/lang/Math;min(II)I", shift = At.Shift.BEFORE))
+    private void hookTabColumnHeight(CallbackInfo ci, @Local(ordinal = 5) LocalIntRef o, @Local(ordinal = 6)LocalIntRef p) {
+        if (!ModuleBetterTab.INSTANCE.getRunning()) {
+            return;
+        }
+
+        int totalPlayers = collectPlayerEntries().size();
+        int columns = 1;
+        int rows = totalPlayers;
+
+        while (rows > ModuleBetterTab.Limits.INSTANCE.getHeight()) {
+            columns++;
+            rows = (totalPlayers + columns - 1) / columns;
+        }
+
+        o.set(rows);
+        p.set(columns);
+    }
+
+    @ModifyArg(method = "render", at = @At(value = "INVOKE", target = "Ljava/lang/Math;min(II)I"), index = 0)
+    private int hookWidth(int width) {
+        return ModuleBetterTab.INSTANCE.getRunning() && ModuleBetterTab.AccurateLatency.INSTANCE.getRunning() ? width + 30 : width;
+    }
+
+    @Inject(method = "renderLatencyIcon", at = @At("HEAD"), cancellable = true)
+    private void hookOnRenderLatencyIcon(DrawContext context, int width, int x, int y, PlayerListEntry entry, CallbackInfo ci) {
+        var accurateLatency = ModuleBetterTab.AccurateLatency.INSTANCE;
+        if (ModuleBetterTab.INSTANCE.getRunning() && accurateLatency.getRunning()) {
+            TextRenderer textRenderer = MinecraftClient.getInstance().textRenderer;
+
+            int latency = MathHelper.clamp(entry.getLatency(), 0, 9999);
+            int color = latency < 150 ? 0x00E970 : latency < 300 ? 0xE7D020 : 0xD74238;
+            String text = latency + (accurateLatency.getSuffix() ? "ms" : "");
+            context.drawTextWithShadow(textRenderer, text, x + width - textRenderer.getWidth(text), y, color);
+            ci.cancel();
+        }
+    }
```

Comment:
We have noticed that you have taken code from the following source: https://github.com/MeteorDevelopment/meteor-client/blob/2025789457e5b4c0671f04f0d3c7e0d91a31765c/src/main/java/meteordevelopment/meteorclient/mixin/PlayerListHudMixin.java#L28

It is very important to get permission and give credit to the original author of the code. In the future, I will not accept pull requests from you if the code is not credited. I will now add the required credits in an additional commit.